packages/clerk-js/src/utils/querystring.ts

```diff
@@ -1,11 +1,47 @@
-import qs from 'qs';
-
 export const getQueryParams = (queryString: string) => {
-  return qs.parse(queryString || '', {
-    ignoreQueryPrefix: true,
-  }) as Record<string, string>;
+  const queryParamsObject: { [key: string]: string | string[] } = {};
+  const queryParams = new URLSearchParams(queryString);
+  queryParams.forEach((value, key) => {
+    if (key in queryParamsObject) {
+      // If the key already exists, we need to handle it as an array
+      const existingValue = queryParamsObject[key];
+      if (Array.isArray(existingValue)) {
+        existingValue.push(value);
+      } else {
+        queryParamsObject[key] = [existingValue, value];
+      }
+    } else {
+      queryParamsObject[key] = value;
+    }
+  });
+  return queryParamsObject as Record<string, string>;
+};
+
+type StringifyQueryParamsOptions = {
+  encoder?: (key: string) => string;
```

Comment:
Let's name this `keyEncoder` as it only deals with object keys and not their values.